ffwd-client/src/main/java/com/spotify/ffwd/v1/Value.java

```diff
@@ -0,0 +1,94 @@
+/*-
+ * -\-\-
+ * FastForward Java Client
+ * --
+ * Copyright (C) 2016 - 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.v1;
+
+import com.google.auto.value.AutoValue;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+
+
+public abstract class Value {
+
+  Value() {}
+
+
+  public static Value doubleValue(double value) {
+    return DoubleValue.create(value);
+  }
+
+  public static Value distributionValue(ByteBuffer byteBuffer) {
```

Comment:
I think this should take ByteString as input instead of ByteBuffer since ByteBuffer is not immutable